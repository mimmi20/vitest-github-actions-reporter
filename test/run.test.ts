import { execa } from 'execa'
import { expect, it } from 'vitest'
import { fileURLToPath } from 'node:url'
import { escapeData, escapeProperty } from './utils'

const currentDir = fileURLToPath(new URL('../', import.meta.url))
const escapedCurrentDirs = [currentDir, currentDir.replace(/\\/g, '/')].flatMap(
  dir => [escapeData(dir), escapeProperty(dir)]
)

const formatMessage = (message: string) => {
  for (const dir of escapedCurrentDirs) {
    message = message.replaceAll(dir, '')
  }
  const messsageList = message.replace(/\r\n/g, '\n').split('\n')
  return [
    messsageList[0], // start group
    ...messsageList.slice(1, -1).sort(),
    messsageList[messsageList.length - 1] // end group
  ]
}

it('should output', async () => {
  const root = new URL('./fixtures', import.meta.url)
  const { stdout } = await execa('npx', ['vitest'], {
    cwd: root,
    env: {
      ...process.env,
      CI: 'true',
      NO_COLOR: 'true'
    }
  }).catch(e => e)

  const message = formatMessage(stdout)
  expect(message).toMatchSnapshot()
}, 50000)
