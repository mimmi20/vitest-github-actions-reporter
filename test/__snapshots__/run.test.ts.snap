// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`should output 1`] = `
[
  "::group::Vitest Annotations",
  "::error title=AssertionError%3A expected 9 to be 10 // Object.is equality,file=test/fixtures/test/fail-test.test.ts,line=6,col=17::  at test/fixtures/test/fail-test.test.ts:6:17",
  "::error title=Error%3A error in file,file=test/fixtures/test/fail-file.test.ts,line=3,col=7::  at test/fixtures/test/fail-file.test.ts:3:7",
  "::error title=Error%3A foo,file=test/fixtures/src/throw.ts,line=2,col=9::  at thrower test/fixtures/src/throw.ts:2:9%0A  at test/fixtures/test/throw.test.ts:5:3",
  "::error title=Error%3A suite error,file=test/fixtures/test/fail-suit.test.ts,line=4,col=9::  at test/fixtures/test/fail-suit.test.ts:4:9",
  "::endgroup::",
]
`;
